src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVMCommon.cpp

```diff
@@ -285,6 +351,24 @@ Value callApi(PatternRewriter &rewriter, Location loc, ModuleOp module,
       ArrayRef<Type>(outputTys), symbolRef, ArrayRef<Value>(params));
 }
 
+// Call a registered API, and check if the return code is zero.
+Value callApiAndCheckReturnCode(PatternRewriter &rewriter, Location loc,
+    ModuleOp module, ApiRegistry registry, API apiId, ArrayRef<Value> params,
+    bool funcCallErrorExit, Value ref) {
+  Value ret = callApi(rewriter, loc, module, registry, apiId, params);
+  if (ref == nullptr) { // if ref is not given, set zero
+    MLIRContext *context = module.getContext();
+    MultiDialectBuilder<MathBuilder> create(rewriter, loc);
+    Type int32Ty = IntegerType::get(context, 32);
+    ref = create.math.constant(int32Ty, 0);
+  }
+  // compare the return value with ref
+  std::string errorMsg =
+      "onnx-mlir: Error in zDNN call(" + apiIdStr(apiId) + "): returned ";
+  equalOrExit(module, rewriter, loc, ref, ret, errorMsg, funcCallErrorExit);
```

Comment:
Since NNPA calls already check the return value, I recommend to make this check optional when we need by providing an option to `onnx-mlir`.